### PR TITLE
New methods: split (Size: 86 B), splitAt (Size: 93 B), splitEvery (Size: 118 B), splitWhen (Size: 124 B)

### PR DESCRIPTION
**Pull request by Kelin2025**
[`R.split`](http://ramdajs.com/docs/#split)  
[`R.splitAt`](http://ramdajs.com/docs/#splitAt)  
[`R.splitWhen`](http://ramdajs.com/docs/#splitWhen)  
[`R.splitEvery`](http://ramdajs.com/docs/#splitEvery)  
@Beraliv performance tests? :smirk:

### CODE REVIEW

File: lib/splitEvery/index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function splitEvery(num, arr) {
+  var res = []
```
Comment (BabkinAleksandr) [inline lib/splitEvery/index.js]: Mutable. Should we always use functional approach, or speed is in priority?

File: lib/splitWhen/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import _curry2 from '../_internal/_curry2'
+
+export default _curry2(function splitWhen(cb, arr) {
+  var left = []
+  while (left.length < arr.length) {
+    if (cb(arr[left.length])) break
+    left.push(arr[left.length])
+  }
+  return [left, arr.slice(left.length)]
```
Comment (Beraliv) [inline lib/splitWhen/index.js]: maybe better have an index to increment and use instead of `left.length`?

File: lib/splitEvery/index.js (hunk 1 above)
Comment (Beraliv) [reply lib/splitEvery/index.js]: We need a balance between them

Comment (Kelin2025) [reply lib/splitEvery/index.js]: Why is this mutable?

File: lib/splitWhen/index.js (hunk 2 above)
Comment (Kelin2025) [reply lib/splitWhen/index.js]: Isn't this shorter?

Comment (Beraliv) [reply lib/splitWhen/index.js]: @Kelin2025 I doubt if it is

**Pull request merged**